crates/extension/src/extension_builder.rs

```diff
@@ -64,11 +64,8 @@ struct CargoTomlPackage {
 }
 
 impl ExtensionBuilder {
-    pub fn new(http_client: Arc<dyn HttpClient>, cache_dir: PathBuf) -> Self {
-        Self {
-            cache_dir,
-            http: http_client,
-        }
+    pub fn new(cache_dir: PathBuf, http: Arc<dyn HttpClient>) -> Self {
```

Comment:
Why did this code need to change?

We should leave it the way it was.

Reply:
I think it is a result of resolving conflict when merging main. Let me change it back to reduce the change sets a bit.